File: examples/x05-1.c
#include <ncurses.h>
#include <stdlib.h>

void testAttr(char *name, int value);

int main()
{
    initscr();

    testAttr("A_NORMAL", A_NORMAL);
    testAttr("A_STANDOUT", A_STANDOUT);
    testAttr("A_UNDERLINE", A_UNDERLINE);
    testAttr("A_REVERSE", A_REVERSE);
    testAttr("A_BLINK", A_BLINK);
    testAttr("A_DIM", A_DIM);
    testAttr("A_BOLD", A_BOLD);
    testAttr("A_PROTECT", A_PROTECT);
    testAttr("A_INVIS", A_INVIS);
    testAttr("A_ALTCHARSET", A_ALTCHARSET);
    testAttr("A_CHARTEXT", A_CHARTEXT);

    testAttr("XXX", A_BOLD + A_UNDERLINE);

    getch();
    endwin();
    return 0;
}

void testAttr(char *name, int value)
{
    
    printw("%s: \t[ ", name);
    attron(value);
    printw("ATTR String test", name, value);
    attroff(value);
    printw("]\t:%x\n", value);
}

File: examples/x05-1.js
"use strict";

/**
 * Example of getstr
 */

$(document).ready(function () {

    x.initscr($("#container"));

    x.printw("Обычная строка...");

    x.attron(x.A_BOLD);
    x.printw("Жирная строка...");
    x.attroff(x.A_BOLD);

    x.printw("Вернулись к обычной строке...");

    x.attron(x.A_BOLD | x.A_UNDERLINE);
    x.printw("Жирная подчеркнутая строка...");
    x.attroff(x.A_BOLD);

    x.printw("Подчеркнутая строка...");

    x.attron(x.A_REVERSE);

    x.printw("Инверсия...");

    x.attron(x.A_BLINK);

    x.printw("Мигающая инверсия...");

    x.attron(x.A_INVIS);
    x.printw("Скрытый текст...");

    x.attrset(x.A_NORMAL);

    x.printw("Вернулись к обычной строке...");

    x.init_pair(1, x.COLOR_LIGHTYELLOW, x.COLOR_DARKRED);
    x.init_pair(2, x.COLOR_LIGHTCYAN, x.COLOR_DARKMAGENTA);

    x.chgat(10, x.A_BOLD, 1);

    x.printw("Цветная строка, но не вся...");

    x.chgat(-1, x.A_BOLD, 2);

    x.printw("Цветная строка, но не вся...");

    x.mvchgat(1,10,5,x.A_BLINK,2);

    x.printw("Ok.......");

    x.refresh();
});

var x = new Webcurses('Example 5');

